_posts/2017-05-10-fastlane-travis-weekly-deploys.md

```diff
@@ -0,0 +1,249 @@
+
+So I automated it. Using Travis CI and fastlane.
+
+<!-- more -->
+
+Travis CI (and Circle CI) now have the ability to run scheduled runs. This is great for one-off tasks like uploading an app to Apple's Testflight on a weekly basis.
+
+I wanted this to exist outside of our current CI environment for two reasons:
+
+* Our CI is already [using AppHub][pr_deploy] to deploy the JavaScript parts of our React Native on a per-commit basis. It's complicated enough as it is, without adding a lot more process.
+* Our CI is currently running on Linux boxes, and so everything is fast and stable. This would force us to use Mac boxes which would have the opposite effect.
+
+The downside is that the process of uploading is not inside the main repo, and can go out of sync. So ideally the new project would be as generic and future-safe as possible.
+
+## Setup
+
+I created a new repo, and added the [usual LICENSE and README][init], then started [working on a PR][pr] that added support for CI to run. Here are the general steps I needed to make work:
+
+* Downloading and setting up the application.
+* Ensuring signing will work.
+* Creating the build and shipping it to Testflight.
+* Notifications that it passed or succeeded.
+
+Finally I needed to document the process, which is what you're reading.
+
+## Downloading and setting up the application
+
+My initial thoughts were to use a submodule, but that option provides little advantage over cloning the repo itself so it's done inline. Our dependencies for the app live in Rubygems (Fastlane/CocoaPods), NPM (React Native) and CocoaPods (Artsy Mobile code), so I use the `before_install` and `before_script` section of the `.travis.yml` to set up our dependencies:
+
+```yml
+# Use a Mac build please
+language: objective-c
+
+# Ensure our dependency managers are up to date
+before_install:
+  - brew install yarn
+  - gem install bundler
+  - bundle update
+
+# Clone emission and setup dependencies
+before_script:
+  - git clone https://github.com/artsy/emission.git
+  - cd emission && yarn install
+  - git pull origin fastlane_match 
+  - cd Example && pod install
+  - cd ../..
+```
+
+Note the `- bundle update`. As both CocoaPods and Fastlane are build tools that have to handle ever changing Apple black-boxes, then locking their versions doesn't make much sense over long time periods for somehting like this. In this case, both of the apps will always be at the latest stable release.
+
+## Ensuring signing will work.
+
+This one is a bit tricker, luckily I've already set up one of our apps to use [fastlane match][] and I can re-use that infrastructure. As it is a closed repo, Travis did not have access to clone the repo. I fixed this by creating an access token for a user with read-only access to our match-codesigning repo, then exposed this as a private environment variable in CI which the Matchfile uses. E.g.
```

Comment:
I actually never got this to work on Travis myself, did you follow a specific guide that was helpful?